sdks/python/apache_beam/runners/interactive/caching/streaming_cache.py

```diff
@@ -153,7 +153,8 @@ def __init__(
     self._coder = coder
     self._labels = labels
     self._is_cache_complete = (
-        is_cache_complete if is_cache_complete else lambda: True)
+        is_cache_complete if is_cache_complete else lambda _: True)
+    self._pipeline_id = labels[-1].split('_')[-1]
```

Comment:
Will it come in as a label?

the ID will be unique for the pipeline object itself. I guess we will not be reusing it anywhere?

Reply:
Yeah the pipeline id is just a pointer to the pipeline, which is guaranteed to be unique and constant.

Reply:
This looks kinda like a hack, and if we change the format of the label, this will break.
Can we at least put the code that encodes and decodes labels in one place, perhaps in a util module, and have this line call the decode function?

Reply:
Done, I added a CacheKey class that can handle the encoding/decoding